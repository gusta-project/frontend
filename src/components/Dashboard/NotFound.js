import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { DashboardLayout as Layout } from 'components/Dashboard';

export class NotFound extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    layoutOptions: PropTypes.object
  };

  render() {
    const { name, layoutOptions } = this.props;

    return (
      <Layout
        pageTitle="Not Found"
        header="Not Found :("
        options={layoutOptions}
      >
        <h3>Oops!</h3>
        I&apos;ve searched <i>long</i> and &quot;hard&quot;, yet I couldn&apos;t
        find a Dashboard Component identified as <strong>{name}</strong> in{' '}
        <code>components/Dashboard/Main.js</code>
      </Layout>
    );
  }
}

export default NotFound;
