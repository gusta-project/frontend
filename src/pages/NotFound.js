import { Helmet } from 'react-helmet';
import React, { Component, Fragment } from 'react';

export default class NotFound extends Component {
  render() {
    return (
      <Fragment>
        <Helmet title="Not Found"></Helmet>
        <h1>Not Found</h1>
        <p>You requested a resource which could not be located.</p>
      </Fragment>
    );
  }
}
