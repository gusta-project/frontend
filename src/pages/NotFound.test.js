import { render } from '@testing-library/react';
import React from 'react';

import NotFound from './NotFound';

describe('<NotFound />', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<NotFound />);

    expect(asFragment()).toMatchSnapshot();
  });
});
