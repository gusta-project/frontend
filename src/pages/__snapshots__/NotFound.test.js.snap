// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      Not Found
    </h1>
    <p>
      You requested a resource which could not be located.
    </p>
  </div>
</DocumentFragment>
`;
